From d6d55d0e0d9737d592125c73cb42bf388d7ae93c Mon Sep 17 00:00:00 2001
From: Silvano di Ninno <silvano.dininno@nxp.com>
Date: Fri, 16 Mar 2018 14:46:28 +0100
Subject: [PATCH 4191/5242] MLK-17674-1: sm_store remove CONFIG_OF

commit  d4d36270e9e8092870cbc6f8626ce1ce9a3796d1 from
https://source.codeaurora.org/external/imx/linux-imx.git

I.MX linux only works with device tree support
No need to keep code without CONFIG_OF

Signed-off-by: Silvano di Ninno <silvano.dininno@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/sm_store.c |   10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/crypto/caam/sm_store.c b/drivers/crypto/caam/sm_store.c
index f5c8f00..7288922 100644
--- a/drivers/crypto/caam/sm_store.c
+++ b/drivers/crypto/caam/sm_store.c
@@ -1019,16 +1019,11 @@ int caam_sm_startup(struct platform_device *pdev)
 	spin_lock_init(&smpriv->kslock);
 
 	/* Create the dev */
-#ifdef CONFIG_OF
 	np = of_find_compatible_node(NULL, NULL, "fsl,imx6q-caam-sm");
 	if (np)
 		of_node_clear_flag(np, OF_POPULATED);
 	sm_pdev = of_platform_device_create(np, "caam_sm", ctrldev);
-#else
-	sm_pdev = platform_device_register_data(ctrldev, "caam_sm", 0,
-						smpriv,
-					sizeof(struct caam_drv_private_sm));
-#endif
+
 	if (sm_pdev == NULL) {
 		kfree(smpriv);
 		return -EINVAL;
@@ -1191,7 +1186,7 @@ void caam_sm_shutdown(struct platform_device *pdev)
 	kfree(smpriv);
 }
 EXPORT_SYMBOL(caam_sm_shutdown);
-#ifdef CONFIG_OF
+
 static void  __exit caam_sm_exit(void)
 {
 	struct device_node *dev_node;
@@ -1248,4 +1243,3 @@ static int __init caam_sm_init(void)
 MODULE_LICENSE("Dual BSD/GPL");
 MODULE_DESCRIPTION("FSL CAAM Secure Memory / Keystore");
 MODULE_AUTHOR("Freescale Semiconductor - NMSG/MAD");
-#endif
-- 
1.7.9.5

